pkg/stanza/fileconsumer/file.go

```diff
@@ -77,6 +97,13 @@ func (m *Manager) Start(persister operator.Persister) error {
 func (m *Manager) Stop() error {
 	m.cancel()
 	m.wg.Wait()
+	if useThreadPool.IsEnabled() {
+		close(m.readerChan)
```

Comment:
In `Start`, it's possible that we return before creating the channel, so we need to check if the channel is nil. This can crash the collector from an otherwise recoverable situation.